nsv13/code/modules/overmap/fighters/fighters.dm

```diff
@@ -420,6 +120,33 @@ You need to fire emag the fighter's IFF board. This makes it list as "ENEMY" on
 		warning_cooldown = TRUE
 		addtimer(VARSET_CALLBACK(src, warning_cooldown, FALSE), 5 SECONDS)
 		return
+	if(damage_amount > max_integrity/2) //Taking a single hit that does over 50% of your max HP is BAD
+		var/destroyed = pick(1, 2, 3, 4) //May as well go AWOL at this point, MAA is going to lynch you
+		for(var/I = 0, I < destroyed, I++)
+			burnout_component()
+		return
+	else if(damage_amount >= 50)
+		if(prob(20))
+			burnout_component()
+		return
+	else if(damage_amount >= 10)
+		if(prob(5))
+			burnout_component()
+		return
+
+/obj/structure/overmap/fighter/proc/burnout_component()
+	if(src == /obj/structure/overmap/fighter/escapepod)
+		return
+	var/list/candidate_list = list()
+	for(var/obj/item/fighter_component/candidate in contents)
+		if(candidate.burntout == FALSE)
+			candidate_list = list(candidate)

```

Comment:
Wouldn't this create a fresh list with only candidate on every loop where the component wasn't burned out? I think this should be "candidate_list += candidate"